test/hotspot/gtest/runtime/test_os.cpp

```diff
@@ -113,39 +116,50 @@ TEST_VM(os, page_size_for_region_unaligned) {
   }
 }
 
-TEST(os, test_random) {
-  const double m = 2147483647;
-  double mean = 0.0, variance = 0.0, t;
-  const int reps = 10000;
-  unsigned int seed = 1;
-
-  // tty->print_cr("seed %ld for %ld repeats...", seed, reps);
-  os::init_random(seed);
-  int num;
-  for (int k = 0; k < reps; k++) {
-    num = os::random();
-    double u = (double)num / m;
-    ASSERT_TRUE(u >= 0.0 && u <= 1.0) << "bad random number!";
-
-    // calculate mean and variance of the random sequence
-    mean += u;
-    variance += (u*u);
+class VM_TestRandom : public VM_GTestExecuteAtSafepoint {
```

Comment:
It took me a while to convince myself that none of the existing uses of os::random() can occur concurrently with a safepoint, but that does seem to be the case.

Reply:
In this test I'm assuming that once this thread gets to a safepoint, it should be the only thing running. Unless maybe some concurrent GC threads.  I don't know what they would be doing for this test case though.